utils/post.js

```diff
@@ -310,39 +310,50 @@ function modifyReactionsPost(post, isAnonimous = true) {
   return newPost;
 }
 
-async function filterFeeds(userId, feeds = [], feed_id = null) {
-  const newResult = [];
-
-  for (const item of feeds || []) {
-    const now = moment().valueOf();
-    const dateExpired = moment(item?.expired_at).valueOf();
-
-    if (dateExpired < now) {
-      if (feed_id) {
-        deleteActivityFromUserFeed('topic', feed_id, item.id);
-      }
-      continue;
+const isExpiredPost = (item, feed_id) => {
+  let result = false;
+  const now = moment().valueOf();
+  const dateExpired = moment(item?.expired_at).valueOf();
+
+  if (dateExpired < now) {
+    if (feed_id) {
+      deleteActivityFromUserFeed('topic', feed_id, item.id);
     }
-    let newItem = {...item};
+    result = true;
+  }
+  return result;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `isExpiredPost` is performing a side effect of deleting an activity from the user feed. This could lead to unexpected behavior as the function name suggests it only checks if a post is expired. Consider separating the deletion logic into its own function for better readability and maintainability.
```diff
const isExpiredPost = (item) => {
  const now = moment().valueOf();
  const dateExpired = moment(item?.expired_at).valueOf();

  return dateExpired < now;
};
 
const deleteExpiredPostFromFeed = (item, feed_id) => {
  if (feed_id) {
    deleteActivityFromUserFeed('topic', feed_id, item.id);
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->